examples/shell.rs

```diff
@@ -30,134 +195,305 @@ impl Context {
+            self.current_volume = 1;
+        } else if line == "C:" || line == "c:" {
+            self.current_volume = 2;
+        } else if line == "D:" || line == "d:" {
+            self.current_volume = 3;
+        } else if line == "dir" {
+            self.dir(Path::new("."))?;
+        } else if let Some(path) = line.strip_prefix("dir ") {
+            self.dir(Path::new(path.trim()))?;
+        } else if line == "tree" {
+            self.tree(Path::new("."))?;
+        } else if let Some(path) = line.strip_prefix("tree ") {
+            self.tree(Path::new(path.trim()))?;
+        } else if line == "stat" {
+            self.stat()?;
+        } else if let Some(path) = line.strip_prefix("cd ") {
+            self.cd(Path::new(path.trim()))?;
+        } else if let Some(path) = line.strip_prefix("cat ") {
+            self.cat(Path::new(path.trim()))?;
+        } else if let Some(path) = line.strip_prefix("hexdump ") {
+            self.hexdump(Path::new(path.trim()))?;
+        } else if let Some(path) = line.strip_prefix("mkdir ") {
+            self.mkdir(Path::new(path.trim()))?;
         } else {
             println!("Unknown command {line:?} - try 'help' for help");
         }
         Ok(())
     }
+
+    /// Resolves an existing directory.
+    ///
+    /// Converts a string path into a directory handle.
+    ///
+    /// * Bare names (no leading `.`, `/` or `N:/`) are mapped to the current
+    ///   directory in the current volume.
+    /// * Relative names, like `../SOMEDIR` or `./SOMEDIR`, traverse
+    ///   starting at the current volume and directory.
+    /// * Absolute, like `1:/SOMEDIR/OTHERDIR` start at the given volume.
+    fn resolve_existing_directory(&mut self, full_path: &Path) -> Result<RawDirectory, Error> {
+        let (dir, fragment) = self.resolve_filename(full_path)?;
+        let mut work_dir = dir.to_directory(&mut self.volume_mgr);
+        work_dir.change_dir(fragment)?;
+        Ok(work_dir.to_raw_directory())
+    }
+
+    /// Either get the volume from the path, or pick the current volume.
+    fn resolve_volume(&self, path: &Path) -> Result<usize, Error> {
+        match path.volume() {
+            None => Ok(self.current_volume),
+            Some('A' | 'a') => Ok(0),
+            Some('B' | 'b') => Ok(1),
+            Some('C' | 'c') => Ok(2),
+            Some('D' | 'd') => Ok(3),
+            Some(_) => Err(Error::NoSuchVolume),
+        }
+    }
+
+    /// Resolves a filename.
+    ///
+    /// Converts a string path into a directory handle and a name within that
+    /// directory (that may or may not exist).
+    ///
+    /// * Bare names (no leading `.`, `/` or `N:/`) are mapped to the current
+    ///   directory in the current volume.
+    /// * Relative names, like `../SOMEDIR/SOMEFILE` or `./SOMEDIR/SOMEFILE`, traverse
+    ///   starting at the current volume and directory.
+    /// * Absolute, like `1:/SOMEDIR/SOMEFILE` start at the given volume.
```

Comment:
Is it intentional to use 1 as a volume label in this example?
Also, a few lines above, N is used as a volume label, but the current implementation only supports A-D.

Reply:
No, that's a left over. Good catch!